### DISCUSSION TITLE
Multiple dbContext support

### DISCUSSION THREAD

**Discussion by anavgil** [ffernandolima/ef-core-data-access#28] [Q&A]
Hi,

I need help about multiple bdcontext support. I have read in documentation that this component has support for multiple dbcontext, but i have tried it with no results.
Could you support about that please?
Thank you in advance.
Great job!!!

**Comment by ffernandolima** [comment]
Hi @anavgil!

Sorry for the late response.

Sure! What's the problem you're facing?

You can open an issue with a repro app and steps, so that I can investigate and then provide you with support regarding the issue.

Thanks.

**Comment by NerdyJ23** [comment]
If you're having a similar issue as to what I was having then I may have a solution. 
My issue was I was not able to select which DbContext was going into the IUnitOfWork, my very basic example is below:

``` csharp
services.AddDbContext<DbContext, ContextA>();
services.AddDbContext<DbContext, ContextB>();

serivces.AddUnitOfWork();
services.AddUnitOfWork<ContextA>();
services.AddUnitOfWork<ContextB>();
```

in my Repository files I was doing the following:
``` csharp
public sealed class Repo  {
    private readonly IUnitOfWork _unitOfWork;

    public Repo(IUnitOfWork unitOfWork) {
        _unitOfWork = unitOfWork;
    }
}
```


Once I looked at the IUnitOfWork interface I noticed there is a generic interface which allows you to select the DbContext, however it was not in the Readme.md examples, so I overlooked it.

Working code was just replacing `IUnitOfWork` in my class with `IUnitOfWork<ContextA>`

``` csharp
public sealed class Repo  {
    private readonly IUnitOfWork<ContextA> _unitOfWork;

    public Repo(IUnitOfWork<ContextA> unitOfWork) {
        _unitOfWork = unitOfWork;
    }
}
```
